Add documentation for monitoring OpenTelemetry Collectors in Fleet (#6135)
elastic/docs-content: reference/fleet/add-otel-collector.md

         receivers: [otlp]
         exporters: [elasticsearch/otel]
       traces:
         receivers: [otlp]
         exporters: [elasticsearch/otel]
     telemetry:
       resource:
         elastic.collector.group_name: "OTel Collector Group"
         elastic.collector.group: "otel-collector-group"
         service.namespace: "otel-collector-group"
         service.name: "otel-collector-group"
         service.instance.id: "${env:HOSTNAME}"
       metrics:
         readers:
           - periodic:
               exporter:
                 otlp:
                   protocol: grpc
                   endpoint: "http://localhost:4317"
       logs:
         processors:
           - batch:
               exporter:
                 otlp:
                   protocol: grpc
                   endpoint: "http://localhost:4317"
       traces:
         processors:
           - batch:
               exporter:
                 otlp:
                   protocol: grpc
                   endpoint: "http://localhost:4317"
   ```
   1. The {{fleet-server}} host URL with the OpAMP endpoint, automatically populated by {{fleet}}.
   2. An enrollment API key, automatically populated by {{fleet}}.

>    3. Skips verification of the {{fleet-server}} TLS certificate. Before deploying to production, replace it with a CA file as described in [Configure TLS for {{fleet-server}} connection](#configure-tls-for-fleet-server-connection).

Reviewer: Instead of "replace it" I would say "adjust the configuration to use a CA file..."

Author: ```suggestion
   3. Skips verification of the {{fleet-server}} TLS certificate. Before deploying to production, adjust the configuration to use a CA file as described in [Configure TLS for {{fleet-server}} connection](#configure-tls-for-fleet-server-connection).
```

   6. The {{es}} endpoint, automatically populated from your default {{fleet}} output.
   7. The {{es}} API key. If you used **Create API key**, {{fleet}} replaces `${API_KEY}` with the generated key. Otherwise, replace `${API_KEY}` manually with your existing encoded key.

   ::::

2. Paste or merge the configuration into your OTel Collector configuration file (for example, `otel.yaml`).
3. Make sure every environment-variable placeholder in the configuration resolves at runtime:
   - `${API_KEY}`: replace it with your encoded {{es}} API key value, or set the `API_KEY` environment variable before starting or restarting the collector.
   - `${env:HOSTNAME}`: make sure `HOSTNAME` is set in the collector's runtime environment, or replace the placeholder with a static identifier.
4. Save your configuration.

:::::

:::::{step} Verify the collector connection

1. Start or restart your OTel Collector with the applied configuration.
2. Return to the {{fleet}} UI. The flyout displays a confirmation message when your collector successfully connects.
3. Your OTel Collector now appears in the **Agents** list.

:::::
::::::

:::{important}

> The generated configuration uses `tls.insecure_skip_verify: true` for the {{fleet-server}} connection. Before deploying to production, replace it with a CA file as described in [Configure TLS for {{fleet-server}} connection](#configure-tls-for-fleet-server-connection).

Reviewer: Same as above regarding "replace it"

Author: ```suggestion
The generated configuration uses `tls.insecure_skip_verify: true` for the {{fleet-server}} connection. Before deploying to production, adjust the configuration to use a CA file as described in [Configure TLS for {{fleet-server}} connection](#configure-tls-for-fleet-server-connection).
```
